File: problemsolving/stack.py
class Stack:
	def __init__(self):
		self.items = []

	def isEmpty(self):
		return len(self.items) == 0
		
	def size(self):
		return len(self.items)
		
	def push(self, item):
		self.items.append(item)
		
	def pop(self):
		return self.items.pop()
		
	def peek(self):
		return self.items[self.size() - 1]


s = Stack();
print("Stack is empty: %s" %(s.isEmpty()))
s.push(4)
print("Size is %d" %(s.size()))
s.push("Carmen")
print(s.pop())
print("Size is %d" %(s.size()))
s.push("Teodor")
print(s.peek())

File: javascript/app/Stack.js
class Node {
    constructor(value) {
        this.value = value;
        this.next = null;
    }
}

class Stack {
    constructor() {
        this.root = null;
        this.size = 0;
    }

    push(value) {
        var newNode = new Node(value);
        this.size += 1;
               
        if (null != this.root) {
            newNode.next = this.root;
        }
        
        this.root = newNode;
    }
        
    pop() {
        if (this.isEmpty()) {
            return null;
        }

        this.size -= 1;
        var node = this.root;
        this.root = this.root.next;
        return node.value;
    }
    
    peek() {
        if (this.isEmpty()) {
            return null;
        }
        
        return this.root.value;       
    }
    
    isEmpty() {
        return this.size == 0;
    }
    
    clear() {
        this.root = null;
        this.size = 0;
    }
        
    print() {
        console.log(stack);
    }
}

module.exports = Stack;
